components/centraldashboard/app/stackdriver_metrics_service.ts

```diff
@@ -0,0 +1,147 @@
+import * as monitoring from '@google-cloud/monitoring';
+
+import {Interval, MetricsService, TimeSeriesPoint} from './metrics_service';
+
+const INSTANCE_METRIC_TYPE_PREFIX = 'compute.googleapis.com/instance';
+const INSTANCE_CPU_UTILIZATION_METRIC_TYPE =
+    `${INSTANCE_METRIC_TYPE_PREFIX}/cpu/utilization`;
+const CONTAINER_METRIC_TYPE_PREFIX = 'container.googleapis.com/container';
+const CONTAINER_CPU_UTILIZATION_METRIC_TYPE =
+    `${CONTAINER_METRIC_TYPE_PREFIX}/cpu/utilization`;
+const CONTAINER_MEMORY_USED_METRIC_TYPE =
+    `${CONTAINER_METRIC_TYPE_PREFIX}/memory/bytes_used`;
+
+export class StackdriverMetricsService implements MetricsService {
+  constructor(
+      private client: monitoring.MetricServiceClient, private projectId: string,
+      private nodeNames: string[]) {}
+
+  async getNodeCpuUtilization(interval: Interval): Promise<TimeSeriesPoint[]> {
+    const response: TimeSeriesPoint[] = [];
+    let filter = `metric.type="${
+        INSTANCE_CPU_UTILIZATION_METRIC_TYPE}" AND metric.label.instance_name = `;
+    if (this.nodeNames.length > 1) {
+      filter += `one_of("${this.nodeNames.join('", "')}")`;
```

Comment:
An alternative approach for readability, if you prefer:

```suggestion
      filter += `one_of(${JSON.stringify(this.nodenames)})`;
```

Reply:
This actually doesn't have the same effect. The latter would return `["node-1","node-2"]` instead of the `node-1", "node-2` that I need inside of the quotes.

Reply:
I see, my bad